### PR TITLE
Update WooCommerce Blocks to version to 11.2.0

### PR DESCRIPTION
**Pull request by tarunvijwani**
### Changes proposed in this Pull Request:

This pull updates the WooCommerce Blocks plugin to 11.2.0. It includes changes from WooCommerce Blocks 11.2.0 and is intended to target WooCommerce 8.3 for release.

Details from all the different releases included in this pull:

## Blocks 11.2.0
* [Release PR](https://github.com/woocommerce/woocommerce-blocks/pull/11052)
* [Testing Instructions](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/testing/releases/1120.md)
* [Release Post](https://developer.woocommerce.com/2023/09/29/woocommerce-blocks-11-2-0-release-notes/)

#### Enhancements

- Remove order and checkout order endpoints experimental flag. ([11022](https://github.com/woocommerce/woocommerce-blocks/pull/11022))
- Ensure the content of the patterns is also AI-generated. ([10997](https://github.com/woocommerce/woocommerce-blocks/pull/10997))
- Product Collection: Transfer layout options from Toolbar to Inspector controls. ([10922](https://github.com/woocommerce/woocommerce-blocks/pull/10922))
- Add pattern chooser in Product Collection. ([10876](https://github.com/woocommerce/woocommerce-blocks/pull/10876))
- Product Gallery: Lock the Sale Badge and the Next/Prev Buttons. ([10869](https://github.com/woocommerce/woocommerce-blocks/pull/10869))
- Refactor Cart and Checkout Page Templates. ([10773](https://github.com/woocommerce/woocommerce-blocks/pull/10773))
-  Blockified Order Confirmation. ([10056](https://github.com/woocommerce/woocommerce-blocks/pull/10056))

#### Bug Fixes

- Product Gallery: Fix the Product Gallery Thumbnails on click. ([11032](https://github.com/woocommerce/woocommerce-blocks/pull/11032))
- WooExpress: Fix Checkout and Cart Blocks Editor Crash. ([11024](https://github.com/woocommerce/woocommerce-blocks/pull/11024))
- Product Gallery: Fix zoom animation on large Image. ([11023](https://github.com/woocommerce/woocommerce-blocks/pull/11023))
- Fix: Password Protection not respected on single product template. ([10999](https://github.com/woocommerce/woocommerce-blocks/pull/10999))
- Product Gallery Pager: Remove the Pager markup if there's only one image. ([10998](https://github.com/woocommerce/woocommerce-blocks/pull/10998))
- Related Products: Hide the block outside of Single Product Template and Single Product block. ([10978](https://github.com/woocommerce/woocommerce-blocks/pull/10978))
- Single Product: Fix the Align setting. ([10977](https://github.com/woocommerce/woocommerce-blocks/pull/10977))
- Hide unexpected bullet point in Product Collection on Storefront. ([10945](https://github.com/woocommerce/woocommerce-blocks/pull/10945))
- Add custom regex for validating Nicaraguan postal codes. ([10928](https://github.com/woocommerce/woocommerce-blocks/pull/10928))
- Update `postcode-validator` to 3.8.15 to validate "new" Taiwanese postcodes. ([10924](https://github.com/woocommerce/woocommerce-blocks/pull/10924))
- BlockTemplatesController: Check that $attributes['theme'] value isset before operating on it. ([10879](https://github.com/woocommerce/woocommerce-blocks/pull/10879))
- Product Gallery: CSS styling tightening up. ([10867](https://github.com/woocommerce/woocommerce-blocks/pull/10867))
- Checkout Block: Prevent changes in the selected shipping method when new rates are added or removed. ([10457](https://github.com/woocommerce/woocommerce-blocks/pull/10457))

### CODE REVIEW

File: plugins/woocommerce/composer.json [hunk 1]
```diff
@@ -17,13 +17,13 @@
 		"php": ">=7.4",
 		"automattic/jetpack-autoloader": "2.11.18",
 		"automattic/jetpack-config": "1.15.2",
-		"automattic/jetpack-connection": "1.51.7",
+		"automattic/jetpack-connection": "^1.57",
```
Comment (tarunvijwani) [inline plugins/woocommerce/composer.json:20]: This change is added to fix the Jetpack dependency error. 
Slack: p1695840907172979/1695821353.214559-slack-C05EVAXCX7U

**Comment by tarunvijwani** [comment]
I will update the Release post URL once the post is published, as it is not yet available.

**Review by roykho — APPROVED** [review]
LGTM!

**Comment by tarunvijwani** [comment]
Updated release post URL. ✅

**Pull request merged**